students/tbrackney/Lesson8/circle_class.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+"""
+File Name: circle.py
+Author: Travis Brackney
+Class: Python 201 - Self paced online
+Date Created 5/16/2018
+Python Version: 3.6.4
+"""
+import math
+
+
+class Circle():
+    """Implements a class of circle objects"""
+    def __init__(self, the_radius):
+        self._r = the_radius
+
+    def __str__(self):
+        return f'Circle with radius: {self._r}'
+
+    def __repr__(self):
+        return f'Circle({self._r})'
+
+    def __add__(self, other):
+        if isinstance(other, Circle):
+            return Circle(self.radius + other.radius)
+        else:
+            raise ValueError(f'{type(other)} cannot be added to a circle')
+
+    def __mul__(self, factor):
+        if factor < 0:
+            raise ValueError('Cannot multiple circle by negative number')
+        else:
+            return Circle(self.radius * factor)
+
+    def __rmul__(self, factor):
+        return Circle(self.radius * factor)
+
+    def __eq__(self, other):
+        return (self.radius == other.radius)
+
+    def __lt__(self, other):
+        return (self.radius < other.radius)
```

Comment:
Remember to wrap your class with the `@functools.total_ordering` decorator so you only need to define `__eq__` and `__lt__` and not all of the other comparison operators